common/src/math.rs

```diff
@@ -105,6 +105,40 @@ pub fn renormalize_isometry<N: RealField>(m: &na::Matrix4<N>) -> na::Matrix4<N>
     translate_along(&direction, boost_length) * rotation.to_homogeneous()
 }
 
+/// returns the result of translating normal vector v0 from p0 to p1
+pub fn translate_normal<N: RealField>(
+    p0: na::Vector4<N>,
+    p1: na::Vector4<N>,
+    v0: na::Vector4<N>,
+) -> na::Vector4<N> {
+    let d = distance(&p0, &p1);
+    // TODO: verfiy this is equvilent to !(d >= na::zero())
+    if na::zero::<N>() > d {
+        return v0;
+    }
+
+    v0 * d.cosh() + p1 * d.sinh()
+}
+
+/// normalizes vector v with repect to translation matrix t
+pub fn normalize_vector<N: RealField>(t: na::Matrix4<N>, v: na::Vector4<N>) -> na::Vector4<N> {
+    let p = t * origin();
+    let m = mip(&v, &v);
+    if m <= na::zero() {
+        return t * na::Vector4::<N>::new(na::one(), na::zero(), na::zero(), na::zero());
+    }
+
+    let v_mag = m.sqrt();
+    let v_norm = v / v_mag;
+
+    (v_norm + p * mip(&p, &v_norm)) * v_mag
```

Comment:
I believe there are some numerical stability reasons for this.

Reply:
Even considering numerical stability, I see no reason why this shouldn't work, nor do I see any reason why the use of `v_mag` would help. It's possible that a previous iteration of this method relied on `v_mag` more, but in the current version, I would expect it to be more of a liability than an asset.

Reply:
@MagmaMcFry Thoughts?

Reply:
Looking at this method again, I don't believe it's quite correct. If `v_mag` squared is 0 or negative, it returns a velocity vector with a speed of 1 in an arbitrary direction, but for a positive `v_mag`, it returns a vector whose length corresponds to the player's speed (which is correct). The suggested replacement with `v + p * mip(&p, &v)` should fix that entirely.

This is almost certainly why the player gets launched slightly in an arbitrary direction in your `jitter_fix` branch, as `apply_forces` doesn't set the velocity to a nonzero value on the first frame.

As for numerical stability, any velocity vector that's close enough to the 3-space that's "orthogonal" (using `mip` instead of `dot`) to the position vector is valid, including the zero vector. Since every iteration of `v + p * mip(&p, &v)` brings `v` closer to that 3-space, and `mip(&p, &v)` is close to zero, when the velocity vector is already correct, I am highly confident that there are no numerical stability issues.